### PR TITLE
Simplify PR requirements while allowing changelog script to apply to any standalone plugin

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

This PR changes and simplifies the requirements for labeling of PRs, while at the same time enhancing the changelog generation script to be flexible to support any standalone plugin.

The proposed changes imply the following:
* The `[Focus] ...` or `Infrastructure` labels are no longer required (and are no longer used by any infrastructure).
* Milestones should exist for every individual standalone plugin rather than only for the Performance Lab plugin. They always need to include a version number.
    * Examples: "Performance Lab 3.0.0", "WebP Uploads 1.0.6".
    * Or we could use the slugs, such as "performance-lab 3.0.0", "webp-uploads 1.0.6", which could be beneficial to further integrate with other plugin specific automation workflows. Either way, this PR does not mandate any particular format, so this could be decided separately.
* The `no milestone` label now actually _means_ "no milestone" rather than "omit this PR from the changelog". The only PRs this should be used for is infrastructure PRs that don't apply to a specific plugin (e.g. build / tooling that affects the entire monorepo).
* A new `skip changelog` label should be used to indicate that a PR should be omitted from the changelog. With this label, it is now possible to assign a target milestone to a PR, without that mandating a changelog entry for it.

The changelog script can be used the same way as before. But now it is flexible to handle _any_ milestone for any standalone plugin rather than only allowing `PL Plugin ...` milestones.

For example:
`npm run changelog -- -m "PL Plugin 3.0.0"` or `npm run changelog -- -m "WebP Uploads 1.0.6"` (given that milestone exists)

In a follow up PR, we could consider enhancing the readme script to also support any standalone plugin, by applying the changes to the correct `readme.txt` file.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Fixes #
 <!--
 For maintainers only, please make sure:
 
-- PR has either `[Focus]` or `Infrastructure` label.
 - PR has a `[Type]` label.
-- PR has a milestone or the `no milestone` label.
+- PR has a plugin specific milestone, or the `no milestone` label if it does not apply to any specific plugin.
```
Comment (westonruter) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- PR has a plugin-specific milestone, or the `no milestone` label if it does not apply to any specific plugin.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Fixes #
...
+- PR has a changelog friendly title, or the `skip changelog` label if it should not be mentioned in the plugin's changelog.
```
Comment (westonruter) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- PR has a changelog-friendly title, or the `skip changelog` label if it should not be mentioned in the plugin's changelog.
```

File: bin/plugin/commands/changelog.js [hunk 3]
```diff
@@ -162,7 +142,14 @@ function getIssueFocus( issue ) {
  * @return {string} The formatted changelog string.
  */
 function formatChangelog( milestone, pullRequests ) {
-	let changelog = '= ' + milestone.replace( 'PL Plugin ', '' ) + ' =\n\n';
+	const version = milestone.match( /\d+\.\d+(\.\d+)?(-[A-Za-z0-9\.]+)?/ );
```
Comment (westonruter) [inline bin/plugin/commands/changelog.js]:
```suggestion
	const version = milestone.match( /\d+\.\d+(\.\d+)?(-[A-Za-z0-9\.]+)?$/ );
```

File: bin/plugin/commands/changelog.js [hunk 4]
```diff
@@ -162,7 +142,14 @@ function getIssueFocus( issue ) {
...
+	if ( ! version ) {
+		throw new Error(
+			`The ${ milestone } milestone does not include a version number.`
```
Comment (felixarntz) [inline bin/plugin/commands/changelog.js]:
```suggestion
			`The ${ milestone } milestone does not end with a version number.`
```

**Comment by felixarntz** [comment]
Workflow-related updates:
* I have created milestones for the upcoming standalone plugin releases (tentatively using slugs and `n.e.x.t` as the version number to use is not always clear for whether it's going to be a minor or a patch release):
    * [auto-sizes n.e.x.t](https://github.com/WordPress/performance/milestone/31)
    * [dominant-color-images n.e.x.t](https://github.com/WordPress/performance/milestone/32)
    * [speculation-rules n.e.x.t](https://github.com/WordPress/performance/milestone/33)
    * [webp-uploads n.e.x.t](https://github.com/WordPress/performance/milestone/34)
    * [embed-optimizer n.e.x.t](https://github.com/WordPress/performance/milestone/35)
    * [optimization-detective n.e.x.t](https://github.com/WordPress/performance/milestone/36)
* I have created the [`skip changelog` label](https://github.com/WordPress/performance/labels/skip%20changelog), please use that as needed from now on while no longer using `no milestone`, unless the PR _really_ doesn't belong to any milestone.

File: bin/plugin/commands/changelog.js [hunk 5]
```diff
@@ -162,7 +142,14 @@ function getIssueFocus( issue ) {
  * @return {string} The formatted changelog string.
  */
 function formatChangelog( milestone, pullRequests ) {
-	let changelog = '= ' + milestone.replace( 'PL Plugin ', '' ) + ' =\n\n';
+	const version = milestone.match( /\d+\.\d+(\.\d+)?(-[A-Za-z0-9\.]+)?$/ );
```
Comment (westonruter) [inline bin/plugin/commands/changelog.js]: No need to escape `.` in a character range. In PhpStorm, this results in a warning:

> Redundant character escape '\.' in RegExp 

So we can remove it:

```suggestion
	const version = milestone.match( /\d+\.\d+(\.\d+)?(-[A-Za-z0-9.]+)?$/ );
```

**Review by westonruter — APPROVED** [review]
LGTM, just one nit.

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks for the PR @felixarntz. The changes to the command should be reflected in the [Performance Lab handbook](https://make.wordpress.org/performance/handbook/performance-lab/releasing-the-plugin/#update-readme-txt).

**Pull request merged**